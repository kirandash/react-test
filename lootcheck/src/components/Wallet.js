import React, { Component } from 'react'
import { connect } from 'react-redux'

// Exporting the non-connected version as well for testing
export class Wallet extends Component {
    render() {
        return (
            <div>
                <h3 className="balance">Wallet Balance: {this.props.balance}</h3>
            </div>
        )
    }
}

// Exporting connected component for use by other components
export default connect(state => {return { balance: state }}, null)(Wallet)
// Note: Can not return an object implicitly. we must use return explicitly
