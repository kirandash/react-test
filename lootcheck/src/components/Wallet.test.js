import { shallow } from 'enzyme';
// Imports the connected Wallet
// import Wallet from './Wallet';
// Import the unconnected version for testing
import { Wallet } from './Wallet'

describe('Wallet', () => {
  const mockDeposit = jest.fn();
  const mockWithdraw = jest.fn();
  const props = { balance: 20, deposit: mockDeposit, withdraw: mockWithdraw }; // action creator will be available to prop
  const wallet = shallow(<Wallet {...props} />);

  it('renders properly', () => {
    expect(wallet.debug()).toMatchSnapshot()
  });

  it('displays the balance from props', () => {
    expect(wallet.find('.balance').text()).toEqual('Wallet Balance: 20')
  });

  it('creates an input to deposit into or withdraw from the balance', () => {
    expect(wallet.find('.input-wallet').exists()).toBe(true);
  });

  describe('when the user types into the wallet input', () => {
    const userBalance = '25';
    
    beforeEach(() => {
        // Simulate change evernt before all test cases in it blocks below
        wallet.find('.input-wallet')
            .simulate('change', { target: { value: userBalance } });
    });

    it('updates the local wallet balance in `state` and converts it to a number', () => {
        expect(wallet.state().balance).toEqual(parseInt(userBalance, 10));
    });

    // Deposit Scenario
    describe('and the user wants to make a deposit', () => {
      // Simulate Click event before each test case
      beforeEach(() => wallet.find('.btn-deposit').simulate('click'));

      it('dispatches the `deposit()` it receives from props with local balance', () => {
        expect(mockDeposit).toHaveBeenCalledWith(parseInt(userBalance, 10));
      });
    });

    // Withdraw Scenario
    describe('and the user wants to make a withdrawal', () => {
      // Simulate Click event before each test case
      beforeEach(() => wallet.find('.btn-withdraw').simulate('click'));

      it('dispatches the `withdraw()` it receives from props with local balance', () => {
        expect(mockWithdraw).toHaveBeenCalledWith(parseInt(userBalance, 10));
      });
    });
  });
});