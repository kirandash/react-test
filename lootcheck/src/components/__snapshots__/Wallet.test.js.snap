// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders properly 1`] = `
"<div>
  <h3 className=\\"balance\\">
    Wallet Balance: 
    20
  </h3>
  <br />
  <input className=\\"input-wallet\\" onChange={[Function (anonymous)]} />
  <button className=\\"btn-deposit\\" onClick={[Function (anonymous)]}>
    Deposit
  </button>
  <button className=\\"btn-withdraw\\" onClick={[Function (anonymous)]}>
    Withdraw
  </button>
</div>"
`;
